### PR TITLE
[Fixtures] Variant name now is concatenated options value.

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Screenshot 2019-07-18 at 08 15 55

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php [hunk 1]
```diff
@@ -391,4 +391,17 @@ private function getRandomValueForProductAttribute(ProductAttributeInterface $pr
...
+
+    private function generateProductVariantName(ProductVariantInterface $variant): string
+    {
+        $variantOptions = $variant->getOptionValues();
+
+        $variantName = '';
+
+        foreach ($variantOptions as $variantOption) {
+            $variantName .= sprintf('%s ', $variantOption->getValue());
+        }
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php]: We could probably use an [array_reduce](https://www.php.net/manual/en/function.array-reduce.php) function for that?

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php [hunk 2]
```diff
@@ -391,4 +392,15 @@ private function getRandomValueForProductAttribute(ProductAttributeInterface $pr
...
+
+    private function generateProductVariantName(ProductVariantInterface $variant): string
+    {
+        return trim(array_reduce(
+            $variant->getOptionValues()->toArray(),
+            static function(string $variantName, ProductOptionValueInterface $variantOption) {
+                return $variantName . sprintf('%s ', $variantOption->getValue());
+            },
+            ''
+    ));
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php]:
```suggestion
	    ));
```

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php [hunk 3]
```diff
@@ -391,4 +392,15 @@ private function getRandomValueForProductAttribute(ProductAttributeInterface $pr
                 throw new \BadMethodCallException();
         }
     }
+
+    private function generateProductVariantName(ProductVariantInterface $variant): string
+    {
+        return trim(array_reduce(
+            $variant->getOptionValues()->toArray(),
+            static function(string $variantName, ProductOptionValueInterface $variantOption) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php]:
```suggestion
            static function(?string $variantName, ProductOptionValueInterface $variantOption) {
```
And removing the third argument? Not sure if it won't be more obvious what is happening here :)

**Comment by Zales0123** [comment]
Thanks, Adam! :1st_place_medal:

**Pull request merged**